drivers/infosilem/room_schedule.cr

```diff
@@ -0,0 +1,70 @@
+require "placeos-driver"
+require "./models"
+
+class Infosilem::RoomSchedule < PlaceOS::Driver
+  descriptive_name "Infosilem Room Schedule Logic"
+  generic_name :RoomSchedule
+  description %(Polls Infosilem Campus Module to expose bookings relevant for the selected System)
+
+  default_settings({
+    infosilem_room_id: "set Infosilem Room ID here",
+    polling_cron:      "*/15 * * * *",
+    debug:             false
+  })
+
+  accessor infosilem : Campus_1
+
+  @room_id : String = "set Infosilem Room ID here"
+  @cron_string : String = "*/15 * * * *"
+  @todays_upcoming_events : Array(Event) = [] of Event
+  @minutes_til_next_event_starts : Int32 | Nil = nil
+  @debug : Bool = false
+
+  def on_load
+    on_update
+  end
+
+  def on_update
+    schedule.clear
+    @debug = setting(Bool, :debug) || false
+    @room_id = setting(String, :infosilem_room_id)
+    @cron_string = setting(String, :polling_cron)
+    schedule.cron(@cron_string) { fetch_and_expose_todays_events }
+  end
+
+  def fetch_and_expose_todays_events
+    today = Time.local.to_s("%Y-%m-%d")
+    todays_events = Array(Event).from_json(fetch_events(today, today))
+    #todays_events = fetch_events(today, today).map { |e| Event.from_json(e) }
+    @todays_upcoming_events = todays_events.select { |e| e.startTime > Time.local }
+    self[:todays_upcoming_events] = @todays_upcoming_events
+
+    return [] of Event if @todays_upcoming_events.empty?
+
+    next_event = @todays_upcoming_events.min_by { |e| e.startTime }
+    update_event_details(next_event)
+    schedule.clear
```

Comment:
Let me know if you think there's a cleaner way of always having the 15min crons (infosilem requests) while only having the 1mins crons when we know that there are >0 events.